File: ruby/ocurrenceCounter.rb
#!/usr/bin/ruby
# frozen_string_literal:true

# create a counter with a given text, then call count_ocurrences with the pattern to search
class OccurrenceCounter
  def initialize(input_text)
    @text = input_text
  end

  # count ocurrences of a pattern in a text
  # @param pattern_to_search [String]
  # @return [Numeric] the count
  def count_occurrences(pattern_to_search)
    @text.scan(/(?=#{pattern_to_search})/).count
  end
end

puts 'ingrese un patrón de texto a buscar'
pattern_to_search = $stdin.gets.chomp
p format('%<number_case>s. %<result>s',
         { number_case: new_case, result: OccurrenceCounter.new(ARGV[0]).count_occurrences(pattern_to_search) })

File: java/OcurrenceCounter.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class OcurrenceCounter {

    public static void main(String[] args) {
        int count = 0;
        int startIndex = 0;
        String inputText=args[0];
        System.out.println("ingrese un patrón de texto a buscar");
        Pattern pattern=Pattern.compile(System.console().readLine());
        Matcher m = pattern.matcher(inputText);
        while(m.find(startIndex)){
            count++;
            startIndex = m.start() + 1;
        }
        System.out.println(count);

    }
}
